File: visualmfrontend/src/app/models/report.ts
import {User} from './user';
import {Material} from './material';

export class Report {
  private id: number;
  private reportMessage: string;
  private user: User;
  private material: Material;
  private solved: boolean;


  constructor(id?: number, reportMessage?: string, solved?: boolean, user?: User, material?: Material) {
    this.id = id;
    this.reportMessage = reportMessage;
    this.solved = solved;
    this.user = user;
    this.material = material;
  }

  getId() {
    return this.id;
  }

  getReportMessage() {
    return this.reportMessage;
  }

  getSolved() {
    return this.solved;
  }

  getUser() : User {
    return this.user;
  }

  getMaterial() : Material {
    return this.material;
  }

  static trueCopy (report):Report {
    return Object.assign(new Report(), report);
  }
}

File: visualmserver/src/main/java/visualmserver/models/Report.java
package visualmserver.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Set;

@Entity
@Table(name = "report")
public class Report {
    @NotNull
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @NotBlank(message = "Report message cannot be empty.")
    @Size(max = 255)
    private String reportMessage;

    @NotNull
    private Boolean solved;

    public Boolean getSolved() {
        return solved;
    }

    public void setSolved(Boolean solved) {
        this.solved = solved;
    }

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "material_sequence_number", nullable = false)
    private Material material;


    public Report() {
    }


    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getReportMessage() {
        return reportMessage;
    }

    public void setReportMessage(String reportMessage) {
        this.reportMessage = reportMessage;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Material getMaterial() {
        return material;
    }

    public void setMaterial(Material material) {
        this.material = material;
    }
}
